File: src/linkedlist/Solution141.java
package linkedlist;// 141. 环形链表

public class Solution141 {
    public boolean hasCycle(ListNode head) {

        ListNode fast = head;
        ListNode slow = head;
        boolean has = false;

        while (fast != null && fast.next != null) {
            fast = fast.next;
            if (slow == fast) {
                has = true;
                break;
            }
            slow = slow.next;
        }

        return has;

    }
}

File: python/linked_list/Solution141.py
# 141. 环形链表
from typing import Optional

from linked_list import ListNode


class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None:
            return False
        if head.next is None:
            return False
        fast = head
        slow = head
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
            if fast == slow:
                return True
        return False
